src/Atmosphere.cpp

```diff
@@ -52,17 +52,17 @@ Atmosphere::Atmosphere(std::shared_ptr<Plugin::Properties> const& pProperties,
   mRenderer.setRayleighAnisotropy(settings.mRayleighAnisotropy);
 
   VistaSceneGraph* pSG = GetVistaSystem()->GetGraphicsManager()->GetSceneGraph();
-  mAtmosphereNode      = pSG->NewOpenGLNode(pSG->GetRoot(), &mRenderer);
+  mAtmosphereNode.reset(pSG->NewOpenGLNode(pSG->GetRoot(), &mRenderer));
   mAtmosphereNode->SetIsEnabled(false);
   VistaOpenSGMaterialTools::SetSortKeyOnSubtree(
-      mAtmosphereNode, static_cast<int>(cs::utils::DrawOrder::eAtmospheres));
+      mAtmosphereNode.get(), static_cast<int>(cs::utils::DrawOrder::eAtmospheres));
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
 Atmosphere::~Atmosphere() {
   VistaSceneGraph* pSG = GetVistaSystem()->GetGraphicsManager()->GetSceneGraph();
-  pSG->GetRoot()->DisconnectChild(mAtmosphereNode);
+  pSG->GetRoot()->DisconnectChild(mAtmosphereNode.get());
```

Comment:
```suggestion
  if (pSG) {
    pSG->GetRoot()->DisconnectChild(mAtmosphereNode.get());
  }
```

Reply:
Are there cases in which the SceneGraph des not exist? I think we never check this pointer...

Reply:
Even though the pointer is probably not null, i would feel better if we always check raw pointers and in this case there are no additional costs in doing so. but it's just a suggestion.

Reply:
In a properly started application it cannot be null. If we start checking every pointer we get from Vista...

In fact you would have to do:
```c++
auto vs = GetVistaSystem();
if(vs)
  auto gm = vs->GetGraphicsManager();
  if (gm)
    sg = gm->GetSceneGraph();
```

... and we have bazillions of such calls  throughout the code base!